nilearn/decoding/searchlight.py

```diff
@@ -162,15 +169,27 @@ def _group_iter_search_light(list_rows, estimator, X, y, groups,
     par_scores : numpy.ndarray
         score for each voxel. dtype: float64.
     """
-    par_scores = np.zeros(len(list_rows))
     t0 = time.time()
+
+    results = dict()
+    first_dim = len(list_rows)
+    
     for i, row in enumerate(list_rows):
         kwargs = dict()
         kwargs['scoring'] = scoring
         kwargs['groups'] = groups
-        par_scores[i] = np.mean(cross_val_score(estimator, X[:, row],
-                                                y, cv=cv, n_jobs=1,
-                                                **kwargs))
+        cross_validation_res = cross_validate(estimator, X[:, row],
+                                              y, cv=cv, n_jobs=1,
+                                              **kwargs)
+        if i == 0:
```

Comment:
I don't understand the logic that is implemented here. It appears that you interpret certain keys differently if the contain 'test' but this feature is not documented...

Reply:
Since I use the dictionary, I wanted to keep the old ```sl.scores_``` format with ndarray as output for those using only the single scoring and not a vector of scores. But using a ndarray as in the previous comment simplifies this logic.